docs/services/rook-services-guide.md

```diff
@@ -0,0 +1,800 @@
+that contain details about how the resources should be provisioned. For
+Yugastore, we will need resource classes that are capable of fulfilling a
+`KubernetesCluster` claim and a `PostgreSQLInstance` claim:
+
+```bash
+cat > classes.yaml <<EOF
+apiVersion: compute.gcp.crossplane.io/v1alpha2
+kind: GKEClusterClass
+metadata:
+  name: standard-gke
+  labels:
+    app: yugastore
+specTemplate:
+  machineType: n1-standard-1
+  numNodes: 1
+  zone: us-central1-b
+  providerRef:
+    name: gcp-provider
+  reclaimPolicy: Delete
+  writeConnectionSecretsToNamespace: crossplane-system
+---
+apiVersion: database.rook.crossplane.io/v1alpha1
+kind: YugabyteClusterClass
+metadata:
+  name: yuga-cluster
+  labels:
+    app: yugastore
+specTemplate:
+  providerRef:
+    name: yugastore-k8s-provider
+  reclaimPolicy: Delete
+  writeConnectionSecretsToNamespace: crossplane-system
+  forProvider:
+    name: hello-ybdb-cluster
+    namespace: rook-yugabytedb
+    master:
+      # Replica count for Master.
+      replicas: 3
+      network:
+        ports:
+          - name: yb-master-ui
+            port: 7000          # default value
+          - name: yb-master-rpc
+            port: 7100          # default value
+      # Volume claim template for Master
+      volumeClaimTemplate:
+        metadata:
+          name: datadir
+        spec:
+          accessModes: [ "ReadWriteOnce" ]
+          resources:
+            requests:
+              storage: 1Gi
+          storageClassName: standard
+    tserver:
+      # Replica count for TServer
+      replicas: 3
+      network:
+        ports:
+          - name: yb-tserver-ui
+            port: 9000
+          - name: yb-tserver-rpc
+            port: 9100          # default value
+          - name: ycql
+            port: 9042          # default value
+          - name: yedis
+            port: 6379          # default value
+          - name: ysql
+            port: 5433          # default value
+      # Volume claim template for TServer
+      volumeClaimTemplate:
+        metadata:
+          name: datadir
+        spec:
+          accessModes: [ "ReadWriteOnce" ]
+          resources:
+            requests:
+              storage: 1Gi
+          storageClassName: standard
+EOF
+
+kubectl apply -f classes.yaml
+```
+
+The `GKEClusterClass` is relatively straightforward in that it configures a
+`GKECluster` and utilizes our previously created GCP `Provider` for connection.
+The `YugabyteClusterClass` is less clear. Starting with the provider, we
+reference a `Provider` that does not currently exist. Because resource classes
+only store configuration data, this is okay as long as the provider exists when
+the class is referenced by a claim. As previously mentioned, this provider will
+be a Kubernetes `Provider` which we will create after the `GKECluster` is
+created and its connection secret is propagated.
+
+The `forProvider` section of the `YugabyteClusterClass` also differs somewhat
```

Comment:
Would move this above the YugabyteClusterClass yaml and add a little more context around using the KubernetesApplication to package the https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/yugabytedb/operator.yaml